metadata-ingestion/src/datahub/ingestion/transformer/add_dataset_tags.py

```diff
@@ -76,7 +76,7 @@ def handle_end_of_stream(
 
         logger.debug("Generating tags")
 
-        for tag_association in self.processed_tags:
+        for tag_association in self.processed_tags.values():
             ids: List[str] = TagUrn.create_from_string(
                 tag_association.tag
             ).get_entity_id()
```

Comment:
while we're here, TagUrn has a `.to_key_aspect()` method that would be way nicer

the mcp can be `MCPW(entityUrn=str(tag_urn), aspect=tag_urn.to_key_aspect())`